src/BaseCollection.php

```diff
@@ -102,7 +102,7 @@ public function map(callable $callback): BaseCollection
 	 */
 	public function filter(callable $callback = null, int $flag = 0): BaseCollection
 	{
-		return new static(array_filter($this->getArrayCopy(), $callback, $flag));
+		return new self(array_filter($this->getArrayCopy(), $callback, $flag));
```

Comment:
This is wrong, as it will return a `BaseCollection` instance when an inherited class instance is expected instead. If the return type hint allows for `static`, then it should be used instead.

Reply:
fixed